src/main/java/org/openrewrite/java/template/processor/RefasterTemplateProcessor.java

```diff
@@ -539,9 +540,29 @@ private String generatePreconditions(List<TemplateDescriptor> beforeTemplates, i
                             if (method.owner.getQualifiedName().toString().startsWith("com.google.errorprone.refaster.")) {
                                 continue;
                             }
-                            String methodName = method.name.toString();
-                            methodName = methodName.equals("<init>") ? "<constructor>" : methodName;
-                            usesVisitors.add("new UsesMethod<>(\"" + method.owner.getQualifiedName().toString() + ' ' + methodName + "(..)\")");
+                            StringBuilder sb = new StringBuilder();
+                            sb.append("new UsesMethod<>(\"");
+                            sb.append(method.owner.getQualifiedName().toString());
+                            sb.append(' ');
+                            if (method.name.toString().equals("<init>")) {
+                                sb.append("<constructor>");
+                            } else {
+                                sb.append(method.name.toString());
+                            }
+                            com.sun.tools.javac.util.List<Type> parameterTypes = method.type.getParameterTypes();
+                            if (parameterTypes.isEmpty()) {
+                                sb.append("()");
+                            } else {
+                                if ((method.flags() & Flags.VARARGS) == 0 && parameterTypes.stream().allMatch(t -> t instanceof Type.JCPrimitiveType || t instanceof Type.ClassType)) {
```

Comment:
There should probably also be a predicate here that checks that the method indeed has overloads.